server/routerlicious/packages/lambdas-driver/src/document-router/documentPartition.ts

```diff
@@ -201,4 +203,43 @@ export class DocumentPartition {
 		this.activityTimeoutTime =
 			activityTime !== undefined ? activityTime : cacluatedActivityTimeout;
 	}
+
+	public pause(offset: number) {
+		if (this.paused) {
+			Lumberjack.info("Doc partition already paused, returning early.", {
+				documentId: this.documentId,
+				tenantId: this.tenantId,
+			});
+			return;
+		}
+		this.paused = true;
+
+		this.q.pause();
+		this.q.remove(() => true); // flush all the messages in the queue since kafka consumer will resume from last successful offset
+
+		if (this.lambda?.pause) {
+			this.lambda.pause(offset);
+		}
+		Lumberjack.info("Doc partition paused", {
+			documentId: this.documentId,
+			tenantId: this.tenantId,
+		});
+	}
+
+	public resume() {
+		if (!this.paused) {
+			Lumberjack.info("Doc partition already resumed, returning early.", {
+				documentId: this.documentId,
+				tenantId: this.tenantId,
+			});
+			return;
+		}
+		this.paused = false;
+
+		this.q.resume();
```

Comment:
It is not needed because once the q is resumed, it will start executing its handler (lambda.handler) as in below screenshot. But I guess I can add it here just in case some lambda needs to implement resume for some logic.

![image](https://github.com/user-attachments/assets/eb0c68f6-7a32-48f7-a617-73d7df8b223a)